contracts/modules/PermissionManager/GeneralPermissionManager.sol

```diff
@@ -101,4 +99,115 @@ contract GeneralPermissionManager is IPermissionManager, Module {
         return allPermissions;
     }
 
+    /**
+    * @notice use to get all delegates
+    * @return address[]
+    */
+    function getAllDelegates() public view returns(address[]) {
+        return allDelegates;
+    }
+
+    /**
+    * @notice use to check if an address is a delegate or not
+    * @param _potentialDelegate the address of potential delegate
+    * @return bool
+    */
+    function isDelegate(address _potentialDelegate) public view returns(bool) {
+
+        require(_potentialDelegate != address(0));
+
+        if (delegateDetails[_potentialDelegate] != bytes32(0)) {
+            return true;
+        }else{
+            return false;
+        }
+    }
+
+    /**
+    * @notice Use to change one or more permissions for a single delegate at once
+    * @param _delegate Ethereum address of the delegate
+    * @param _multiModules Multiple module matching the multiperms, needs to be same length
+    * @param _multiPerms Multiple permission flag needs to be changed
+    * @return nothing
+    */
+    function changePermissionBulk(
+        address _delegate,
+        address[] _multiModules,
+        bytes32[] _multiPerms
+    )
+    external
+    withPerm(CHANGE_PERMISSION)
+    {
+        require(delegateDetails[_delegate] != bytes32(0), "Delegate details not set");
+        require(_multiModules.length != 0 && _multiPerms.length !=0);
+        
+        for(uint8 i=0;i<_multiPerms.length;i++){
+            bool _currentPerm = !perms[_multiModules[i]][_delegate][_multiPerms[i]];
+            perms[_multiModules[i]][_delegate][_multiPerms[i]] = _currentPerm;
```

Comment:
as per the above declaration 
         
       // Mapping used to hold the permissions on the modules provided to delegate, delegate add => modules add => permission uint8 => bool 
    mapping (address => mapping (address => mapping (bytes32 => bool))) public perms;

it should be `perms[_delegate][multiModules[i]][_multiPerms[i]]` or I misunderstand something

Reply:
the mapping comment is incorrect, it is module => delegate => perm. 

code is correct, will update the comment.